analyzers/src/SonarAnalyzer.CSharp/Rules/EqualityOnFloatingPoint.cs

```diff
@@ -85,12 +85,21 @@ private static void CheckEquality(SonarSyntaxNodeReportingContext context)
                 && IsFloatingPointNumberType(container)
                 && EqualityOperators.Contains(equalitySymbolName))
             {
-                var messageEqualityPart = GetMessageEqualityPart(equals.IsKind(SyntaxKind.EqualsExpression));
-
-                context.ReportIssue(Diagnostic.Create(rule, equals.OperatorToken.GetLocation(), messageEqualityPart));
+                var messageEqualityPart = MessageEqualityPart(equals.IsKind(SyntaxKind.EqualsExpression));
+                var proposed = ProposedMessage(context, equals.Right) ?? ProposedMessage(context, equals.Left) ?? "a range";
+                context.ReportIssue(Diagnostic.Create(Rule, equals.OperatorToken.GetLocation(), messageEqualityPart, proposed));
             }
         }
 
+        private static string ProposedMessage(SonarSyntaxNodeReportingContext context, ExpressionSyntax expression) =>
+            expression is MemberAccessExpressionSyntax memberAccess
+            && memberAccess.GetName() is var memberName
+            && FloatingPointEpsilonDependantMembers.TryGetValue(memberName, out var proposedMethod)
+            && context.SemanticModel.GetSymbolInfo(memberAccess.Expression).Symbol is ITypeSymbol type
+            && IsFloatingPointNumberType(type)
+            ? $"{type}.{proposedMethod}"
```

Comment:
Use [`ToMinimalDisplayString`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.isymbol.tominimaldisplaystring?view=roslyn-dotnet-3.9.0)

Reply:
I would scope the alias definition to a single test case. That way the problems go away.
There might also be an option in the formatting setting on how to handle aliases so you could in theory turn off the aliasing but I wouldn't do that. That alias stuff is really unlikely to be an issue because no sane person would alias System.Double which already has an alias (the keyword double). The alias stuff makes more sense for [`Vector<T>`](https://learn.microsoft.com/en-us/dotnet/api/system.numerics.vector-1?view=net-7.0) and the like.